src/main/java/com/github/kay9/dragonmounts/dragon/ai/DragonAi.java

```diff
@@ -0,0 +1,197 @@
+
+    public static Brain.Provider<TameableDragon> brainProvider()
+    {
+        return Brain.provider(MEMORY_TYPES, SENSOR_TYPES);
+    }
+
+    public static Brain<?> makeBrain(Brain<TameableDragon> brain)
+    {
+        initCoreActivity(brain);
+        initIdleActivity(brain);
+        initFightActivity(brain);
+        initAvoidActivity(brain);
+        initSitActivity(brain);
+        brain.setCoreActivities(ImmutableSet.of(Activity.CORE));
+        brain.setDefaultActivity(Activity.IDLE);
+        brain.useDefaultActivity();
+        return brain;
+    }
+
+    private static void initCoreActivity(Brain<TameableDragon> brain)
+    {
+        brain.addActivity(Activity.CORE, 0, ImmutableList.of(
+                new Swim(0.8f),
+                new SitWhenOrderedTo(),
+                new FightWithOwner(),
+                new LookAtTargetSink(45, 90),
+                new LiftOffIfTargetIsHighEnough(),
+                new LiftOffIfStuck(),
+                new MoveToTargetSink()));
+    }
+
+    private static void initIdleActivity(Brain<TameableDragon> brain)
+    {
+        brain.addActivity(Activity.IDLE, 10, ImmutableList.of(
+                new AnimalMakeLove(DMLRegistry.DRAGON.get(), 1.0f),
+                new TeleportToOwnerIfFarEnough(),
+                new SetWalkTargetToOwnerIfFarEnough(1.0f),
+                SetEntityLookTargetSometimes.create(EntityType.PLAYER, 10f, UniformInt.of(30, 60)), // todo was never a fan of looking exclusively at players. Should dragons look at everything?
+                StartAttacking.create(DragonAi::canAttackRandomly, DragonAi::findNearestValidAttackTarget),
+                getIdleMovementBehaviors()));
+    }
+
+    private static void initFightActivity(Brain<TameableDragon> brain)
+    {
+        brain.addActivityAndRemoveMemoryWhenStopped(Activity.FIGHT, 10, ImmutableList.of(
+                StopAttackingIfTargetInvalid.create(),
+                SetWalkTargetFromAttackTargetIfTargetOutOfReach.create(1.0F),
+                MeleeAttack.create(40),
+                EraseMemoryIf.create(DragonAi::wantsToStopFighting, MemoryModuleType.ATTACK_TARGET)
+        ), MemoryModuleType.ATTACK_TARGET);
+    }
+
+    private static void initAvoidActivity(Brain<TameableDragon> brain)
+    {
+        brain.addActivityAndRemoveMemoryWhenStopped(Activity.AVOID, 10, ImmutableList.of(
+                SetWalkTargetAwayFrom.entity(MemoryModuleType.AVOID_TARGET, 1.5f, 16, false),
+                SetEntityLookTargetSometimes.create(EntityType.PLAYER, 10.0F, UniformInt.of(30, 60)),
+                getIdleMovementBehaviors(), // todo is this necessary?
```

Comment:
It's a bit weird if I remember correctly. If this is in core behaviors it supersedes other things, so it has to be added to any activity where the dragon may be idling. Without it, the dragon will basically freeze up until the avoid target is cleared or approaches.

You could just clear the avoid target as soon as you're outside a certain range, but then the moment it crosses that it will not flee unless it's hit again.

Reply:
Yeah I haven't done any testing myself, or even familiarized myself with it, just to me felt odd to duplicate things. I understand